client/src/index.jsx

```diff
@@ -0,0 +1,88 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import axios from 'axios';
+import $ from 'jquery';
+import ProductList from './components/ProductList.jsx';
+
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      products: [],
+    };
+    this.getNextProducts = this.getNextProducts.bind(this);
+    this.getPreviousProducts = this.getPreviousProducts.bind(this);
+    (this.index = 0), (this.size = 6), (this.data = []);
+  }
+
+  componentDidMount() {
+    this.getProducts();
+  }
+
+  getProducts() {
+    const self = this;
+    axios
+      .get('/products')
+      .then((response) => {
+        self.data = response.data;
```

Comment:
There might be some scaling issues with this implementation since you are grabbing all of the data at the start. Also, do you have plans on having it load different data depending on the product you are currently on?

Reply:
This is a great point.  I'll have to refactor it when I implement a data generator.  I think the plan is to do that, but focusing on tests first and seeing how much time I have left.  Also still thinking about how I would implement the relation to each individual product (maybe random?).